src/Files.App/Actions/FileSystem/RenameAction.cs

```diff
@@ -33,15 +33,22 @@ public RenameAction()
 		}
 
 		public Task ExecuteAsync(object? parameter = null)
-		{
-			context.ShellPage?.SlimContentPage?.ItemManipulationModel.StartRenameItem();
+		{	
+			if (context.SelectedItems.Count == 1)
+			{
+				context.ShellPage?.SlimContentPage?.ItemManipulationModel.StartRenameItem();
```

Comment:
What are your thoughts on moving the logic to within `StartRenameItem`?

Reply:
We commited that change now. Thank you for the feedback

Reply:
Thanks!